test/terra/backends/qasm_simulator/qasm_cliffords.py

```diff
@@ -237,6 +237,26 @@ def test_swap_gate_nondeterministic_default_basis_gates(self):
         self.assertSuccess(result)
         self.compare_counts(result, circuits, targets, delta=0.05 * shots)
 
+    # ---------------------------------------------------------------------
+    # Test pauli gate
+    # ---------------------------------------------------------------------
+    def test_pauli_gate_deterministic_default_basis_gates(self):
+        """Test multipauli-gate circuits compiling to backend default basis_gates."""
+        backends_to_test = ['statevector', 'statevector_gpu', 'statevector_thrust']
+        shots = 100
+        circuits = ref_1q_clifford.pauli_gate_circuits_deterministic(
+            final_measure=True)
+        targets = ref_1q_clifford.pauli_gate_counts_deterministic(shots)
+        basis_gates = None
+        if 'method' not in self.BACKEND_OPTS or self.BACKEND_OPTS['method'] in backends_to_test:
+            basis_gates = ['pauli', 'h', 'z'] # implemented only for statevector for now
+        job = execute(circuits, self.SIMULATOR, shots=shots,
```

Comment:
I don't think this is testing the gate, since you haven't added it to the basis gates it will be getting unrolled to regular I, X, Y, Z instructions.

Reply:
I actually verified this explicitly when I wrote it - changes to `apply_pauli` in the cpp code casued test failures. Note I pass the basis gates when running the job:

```
job = execute(circuits, self.SIMULATOR, shots=shots,
                      basis_gates=basis_gates,
                      backend_options=self.BACKEND_OPTS)
```

Reply:
Anyway, I changed the code the be cleaner - I obtain the basis gates list directly from the backend's `_method_configuration` method.